### PR TITLE
refactoring administrative ui implementation and removing unneeded styles...

### PR DESCRIPTION
**Pull request by hclarkEnq**
refactoring administrative ui implementation and removing unneeded styles...

## Description

Create Inputs for Administrative section with the following fields:
as of
Comments
Figma link: https://www.figma.com/design/3BUUZ6e0kGbekLrA77SFDi/NBS-7.x--%3E-Patient-Data-Entry-1.1?node-id=1082-117950&t=GBCBmzaiHTcslpDX-4Connect your Figma account 
Acceptance Criteria
Administrative information as of- This is mandatory field and error has to be displayed when there is no data.

## Tickets

* [Jira Ticket](https://cdc-nbs.atlassian.net/browse/CNFT1-2749)

## Checklist before requesting a review
- [x] PR focuses on a single story
- [x] Code has been fully tested to meet acceptance criteria
- [x] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [x] All new functions/classes/components reasonably small
- [x] Functions/classes/components focused on one responsibility
- [x] Code easy to understand and modify (clarity over concise/clever)
- [x] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [x] PR does not contain hardcoded values (Uses constants)
- [x] All code is covered by unit or feature tests

### CODE REVIEW

File: apps/modernization-ui/src/apps/patient/add/extended/inputs/administrative/Administrative.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { internalizeDate } from 'date';
+import { AdministrativeEntryFields } from 'apps/patient/profile/administrative/AdministrativeEntryFields';
+import { AdministrativeEntry } from 'apps/patient/profile/administrative/AdministrativeEntry';
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/patient/add/extended/inputs/administrative/Administrative.tsx:3]: You're going to want to use this [AdministrativeEntry](https://github.com/CDCgov/NEDSS-Modernization/blob/main/apps/modernization-ui/src/apps/patient/data/entry.ts#L11) that was created for the new form object.  This also means that we cannot resuse the `AdministrativeEntryFields` component from Patient Profile.  The main reason being the ability to support `Selectable`

File: apps/modernization-ui/src/apps/patient/add/extended/inputs/administrative/Administrative.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    asOf: internalizeDate(new Date()),
+    comment: ''
+};
+
+type Props = {
+    onChange: (data: AdministrativeEntry) => void;
+};
+
+export const Administrative = ({ onChange }: Props) => {
+    const form = useForm<AdministrativeEntry>({
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/patient/add/extended/inputs/administrative/Administrative.tsx:19]: issue:  This is not a multi-valued area so it's part of the main form.  Only the multi-value entry areas require the sub-form.

File: apps/modernization-ui/src/apps/patient/add/extended/AddPatientExtendedForm.tsx [hunk 3]
```diff
@@ -10,46 +10,42 @@ import { RaceMultiEntry } from './inputs/race/RaceMultiEntry';
...
         <>
             <div className={styles.addPatientForm}>
                 <FormProvider {...form}>
                     <div className={styles.formContent}>
-                        <AdministrativeSingleEntry
+                        <Administrative
                             onChange={(data) => {
-                                form.setValue('administrative', data);
+                                form.setValue('asOf', data.asOf);
+                                form.setValue('comments', data.comment);
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/patient/add/extended/AddPatientExtendedForm.tsx:47]: issue:  There is no need to keep two forms in sync, the main form can be used here

**Review by adamloup-enquizit — CHANGES_REQUESTED** [review]
The[ form object](https://github.com/CDCgov/NEDSS-Modernization/blob/6f51c501035967dd5e47709237d674dcc9daf494/apps/modernization-ui/src/apps/patient/add/extended/entry.ts#L15) was defined in [CNFT1-2995](https://github.com/CDCgov/NEDSS-Modernization/pull/1788) and should be used for this page.

File: apps/modernization-ui/src/apps/patient/add/extended/inputs/administrative/Administrative.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import { render } from '@testing-library/react';
+import { Administrative } from './Administrative';
+import { FormProvider, useForm } from 'react-hook-form';
+
+const mockOnChange = jest.fn();
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/patient/add/extended/inputs/administrative/Administrative.spec.tsx:5]: question:  Is this still needed?

File: apps/modernization-ui/src/apps/patient/add/extended/inputs/administrative/Administrative.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+                />
+
+                <Controller
+                    control={control}
+                    name="administrative.comment"
+                    rules={{
+                        ...maxLengthRule(2000)
+                    }}
+                    render={({ field: { onBlur, onChange, value, name }, fieldState: { error } }) => (
+                        <>
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/patient/add/extended/inputs/administrative/Administrative.tsx:45]: question:  was this left in intentionally?

File: apps/modernization-ui/src/apps/patient/add/extended/inputs/administrative/Administrative.module.scss [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+        align-items: center;
+        background-color: colors.$base-white;
+
+        @extend %thin-bottom;
+    }
+
+    .form {
+        background-color: colors.$base-white;
+    }
+}
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/patient/add/extended/inputs/administrative/Administrative.module.scss:23]: suggestion:  This is going to be common to all of the section in this page and could be it's own component.

File: apps/modernization-ui/src/apps/patient/profile/administrative/AdministrativeEntryFields.tsx [hunk 7]
```diff
@@ -31,7 +32,6 @@ export const AdministrativeEntryFields = () => {
                 control={control}
                 name="comment"
                 rules={{
-                    required: { value: true, message: 'Comments are required.' },
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/patient/profile/administrative/AdministrativeEntryFields.tsx:34]: issue:  revert this change as it affects the patient profile

**Review by adamloup-enquizit — CHANGES_REQUESTED** [review]
It looks like there are some left over changes to patient profile that need to be reverted

**Review by alaapbharadwaj — DISMISSED** [review]

File: apps/modernization-ui/src/apps/patient/add/extended/inputs/administrative/Administrative.tsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+import { Input } from 'components/FormInputs/Input';
+import { maxLengthRule } from 'validation/entry';
+import { DatePickerInput } from 'components/FormInputs/DatePickerInput';
+import { Wrapper } from '../../wrapper/Wrapper';
+
+export const Administrative = () => {
+    const { control } = useFormContext<{ administrative: AdministrativeEntry }>();
+
+    return (
+        <Wrapper title="Administrative" id="administrative">
```
Comment (alaapbharadwaj) [inline apps/modernization-ui/src/apps/patient/add/extended/inputs/administrative/Administrative.tsx:12]: issue: i think the id should be something like "section-Administrative" so the right hand side nav can link to the right section. Check AddPatientExtendedNav.tsx for the exact id value

**Review by alaapbharadwaj — CHANGES_REQUESTED** [review]

**Review by alaapbharadwaj — APPROVED** [review]

**Review by mpeels — APPROVED** [review]

**Review by alaapbharadwaj — APPROVED** [review]

**Review by adamloup-enquizit — APPROVED** [review]
👍

**Pull request merged**